File: gwan/gwan_linux64-bit/0.0.0.0_8088/#0.0.0.0/csp/json.cpp
// Test type 1: JSON serialization
// http://localhost:8088/?json.cpp
#pragma link "json"
#include "gwan.h"
#include "gwan-helper.h"
#include "json-obj.h"

class Benchmark {
  protected:
    Http res;
  public:
    Benchmark(int argc,char** argv) : res(argc,argv) {
      res.set_content_type("application/json");
    }
    int exec() {
      JsonObj obj;
      obj.add("message","Hello, World!");
      res << obj.to_s();
      return res.end();
    }
    virtual ~Benchmark() {}
};

int main(int argc,char*argv[]) {
  try {
    Benchmark b(argc,argv);
    return b.exec();
  } catch(...) {
    return 500;
  }
}

File: frameworks/TypeScript/typescript-rest/src/controllers/json.ts
import { GET, Path } from "typescript-rest";

interface IResult {
  message: string;
}

@Path("/json")
export default class Json {
  /**
   * Implements the `JSON Serialization` test type. Under
   * the hood, Express should have serialized the result
   * with `res.json`.
   */

  @GET
  json(): IResult {
    return { message: "Hello, World!" };
  }
}
